authors/apps/articles/views.py

```diff
@@ -92,3 +90,83 @@ def update(self, request, slug,*args,**kwargs):
         serializer.is_valid(raise_exception=True)
         serializer.save()
         return Response(serializer.data)
+
+
+class ShareFacebookView(ListCreateAPIView):
+    """creating links for sharing articles via facebook
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleShareLinkRenderer, )
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get("slug")
+        return get_object_or_404(Article, slug=slug)
+
+    def post(self, request, slug, *args, **kwargs):
+        self.get_object(slug)
+        current_url = 'http://{}'.format(get_current_site(request))
+        route = 'api/articles'
+
+        url = "{}/{}/{}".format(current_url, route, slug)
+
+        facebook_url = "https://www.facebook.com/sharer/sharer.php?u="
+        share_link = facebook_url + url
+
+        return Response(share_link)
+
+
+class ShareTwitterView(ListCreateAPIView):
+    """creating links for sharing articles via facebook
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleShareLinkRenderer, )
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get("slug")
+        return get_object_or_404(Article, slug=slug)
+
+    def post(self, request, slug, *args, **kwargs):
+        self.get_object(slug)
+        current_url = 'http://{}'.format(get_current_site(request))
+        route = 'api/articles'
+
+        url = "{}/{}/{}".format(current_url, route, slug)
+
+        Twitter_url = "https://twitter.com/home?status="
+        share_link = Twitter_url + url
+
+        return Response(share_link)
+
+
+class ShareEmailView(ListCreateAPIView):
+    """creating links for sharing articles via facebook
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleEmailLinkRenderer, )
+    serializer_class = EmailSerializer
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get("slug")
+        return get_object_or_404(Article, slug=slug)
+
+    def post(self, request, slug, *args, **kwargs):
+        reciever_mail = request.data
+        serializer = self.serializer_class(data=reciever_mail)
+        serializer.is_valid(raise_exception=True)
+        self.get_object(slug)
+        current_url = 'http://{}'.format(get_current_site(request))
+        route = 'api/articles'
+
+        url = "{}/{}/{}".format(current_url, route, slug)
+
+        from_email = default.DEFAULT_FROM_EMAIL
+        to_mail = [reciever_mail]
+        subject = "Authors Haven Article"
+        body = "Click here to view article {}".format(url)
+        send_mail(subject, body, from_email, to_mail, fail_silently=False)
+        resp = "email sent successfully"
```

Comment:
@frankopkusianwar could you return a more meaningful message in the response.
It should describe the operation that a user has done